### PR TITLE
sample_misc: Addin WAYLAND DMABUF support to render NV12 on Wayland.

### PR DESCRIPTION
**Pull request by UngTengEn**
Similar fix for [this](https://github.com/Intel-Media-SDK/MediaSDK/pull/2803).
However, this version will use wayland-scanner to auto generate the wayland binding code.

### CODE REVIEW

**Review by dmitryermilov — DISMISSED** [review]
LGTM
Appreciate a review from @onabiull and @dvrogozh

File: samples/sample_misc/wayland/src/class_wayland.cpp [hunk 1]
```diff
@@ -347,20 +351,53 @@ struct wl_buffer * Wayland::CreatePrimeBuffer(uint32_t name
...
-            , width
-            , height
-            , format
-            , offsets[0]
-            , pitches[0]
-            , offsets[1]
-            , pitches[1]
-            , offsets[2]
-            , pitches[2]);
+#if defined(WAYLAND_LINUX_DMABUF_SUPPORT)
```
Comment (dvrogozh) [inline samples/sample_misc/wayland/src/class_wayland.cpp:355]: Hm... Should we make this unconditional and require this feature upfront on configuration level? @dmitryermilov, @onabiull : how do you think?

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: samples/sample_misc/wayland/src/class_wayland.cpp (hunk 1 above)
Comment (dmitryermilov) [reply samples/sample_misc/wayland/src/class_wayland.cpp:355]: Nice to have it by default of course but I'm not sure about availability of wayland-scanner>=1.15, wayland-protocols>=1.15 is OS distributions.

File: samples/sample_misc/wayland/CMakeLists.txt [hunk 2]
```diff
@@ -12,4 +12,35 @@ set(DEPENDENCIES  libva libdrm wayland-client)
...
+if (WAYLAND_LINUX_DMABUF_XML_PATH AND WAYLAND_SCANNER_BINARY_PATH)
+  execute_process(
+	  COMMAND "${WAYLAND_SCANNER_BINARY_PATH}"
+              "client-header" "${WAYLAND_LINUX_DMABUF_XML_PATH}"
+              "samples/sample_misc/wayland/linux-dmabuf-unstable-v1.h"
+      WORKING_DIRECTORY "${CMAKE_BINARY_DIR}"
+      RESULT_VARIABLE WAYLAND_SCANNER_RESULT)
+  if (WAYLAND_SCANNER_RESULT)
+    message(ERROR "Failed to generate linux-dmabuf-unstable-v1.h")
+    return()
```
[the remark this answers is not in the document]
Comment (dmitryermilov) [reply samples/sample_misc/wayland/CMakeLists.txt:24]: @UngTengEn , could you fix it?

File: samples/sample_misc/wayland/CMakeLists.txt [hunk 3]
```diff
@@ -12,4 +12,35 @@ set(DEPENDENCIES  libva libdrm wayland-client)
...
+  endif()
+
+  execute_process(
+      COMMAND "${WAYLAND_SCANNER_BINARY_PATH}"
+              "private-code" "${WAYLAND_LINUX_DMABUF_XML_PATH}"
+              "samples/sample_misc/wayland/linux-dmabuf-unstable-v1.c"
+      WORKING_DIRECTORY "${CMAKE_BINARY_DIR}"
+      RESULT_VARIABLE WAYLAND_SCANNER_RESULT)
+  if (WAYLAND_SCANNER_RESULT)
+    message(ERROR "Failed to generate linux-dmabuf-unstable-v1.c")
```
[the remark this answers is not in the document]
Comment (dmitryermilov) [reply samples/sample_misc/wayland/CMakeLists.txt:34]: @UngTengEn , could you fix it?

File: samples/sample_misc/wayland/src/class_wayland.cpp (hunk 1 above)
Comment (dmitryermilov) [reply samples/sample_misc/wayland/src/class_wayland.cpp:355]: Not  blocker for merge, IMO

File: samples/sample_misc/wayland/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (UngTengEn) [reply samples/sample_misc/wayland/CMakeLists.txt]: Changed, there is new patch.

File: samples/sample_misc/wayland/CMakeLists.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (UngTengEn) [reply samples/sample_misc/wayland/CMakeLists.txt]: Changed, there is new patch.

**Review by dmitryermilov — APPROVED** [review]

**Pull request merged**